server/src/modules/traverseModule/filesFinder.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+import path from 'path';
+import imageType from 'image-type';
+
+export const getAllFiles = (baseFolder: string, recursive: boolean = true, filesList: string[] = []): string[] => {
```

Comment:
Can you make this recursive function yield results every time the array reaches a size of argument N?
reference: 
https://stackoverflow.com/questions/32789593/recursive-generators-in-javascript
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*
https://dev.to/hemaka/lazy-recursion-using-javascript-generators-4f5l

something a little bit like this could work
```
let files = getAllFiles(baseFolder, recursive, n);
```

```
export const getAllFiles = (baseFolder: string, recursive: boolean = true, fliesBatch: number = 10, filesList: string[] = []): string[] => {
      ...
      filesList.push(...regularFiles);
      if(filesList.lenght() >= n){
            yield filesList.slice(0, n)
            filesList = filesList.slice(n)
      }
      if (recursive){
		absoluteFiles.forEach((file: any) => {
			if (fs.statSync(file).isDirectory()){
				yield * getAllFiles(file, recursive, filesList);
			}
		});
       }
       yield filesList
}

Reply:
I've made generator returning single element instead of whole batch, because code is simpler and better. You still can use it to create batch getting `n` elements from generator.

Reply:
You can remove old functions and keep generators